src/main/java/org/folio/service/inventory/InventoryManager.java

```diff
@@ -746,35 +739,11 @@ private Future<List<JsonObject>> getContributorNameTypeByIds(List<String> ids, R
 
   public Future<JsonObject> getEntryId(String entryType, ErrorCodes errorCode, RequestContext requestContext) {
     Promise<JsonObject> promise = Promise.promise();
-    var vertx = requestContext.getContext().owner();
-    SharedData sharedData = vertx.sharedData();
-    String lockName = TenantTool.tenantId(requestContext.getHeaders()) + "." + entryType;
-
-    sharedData.getLock(lockName, lockResult -> {
-      if (lockResult.succeeded()) {
-        logger.debug("Got lock {}", lockName);
-        Lock lock = lockResult.result();
-        try {
-          vertx.setTimer(30000, timerId -> releaseLock(lock, lockName));
-          getAndCache(entryType, requestContext).onComplete(result -> {
-            releaseLock(lock, lockName);
-            if (result.succeeded()) {
-              promise.complete(result.result());
-            } else {
-              getEntryTypeValue(entryType, requestContext)
-                .onSuccess(entryTypeValue -> promise.fail(new HttpException(500, buildErrorWithParameter(entryTypeValue, errorCode))));
-            }
-          });
-        } catch (Exception e) {
-          releaseLock(lock, lockName);
-          promise.fail(e);
-        }
-      } else {
-        logger.error("Error acquiring lock");
-        promise.fail(
-            new io.vertx.ext.web.handler.HttpException(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(), lockResult.cause().getMessage()));
-      }
-    });
+
+    getAndCache(entryType, requestContext)
+      .onSuccess(promise::complete)
+      .onFailure(t -> getEntryTypeValue(entryType, requestContext)
+        .onSuccess(entryTypeValue -> promise.fail(new HttpException(500, buildErrorWithParameter(entryTypeValue, errorCode)))));
```

Comment:
In case if getEntryTypeValue() completes with error - our promise will be never failed that can cause some leaks.
I propose to change .onSuccess to .onComplete to fail promise in any case.
If we able to return entry type value - we can return specific error message with this param, or without it in case of failures

Reply:
good point. thanks